src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -257,14 +257,26 @@ class device {
         sensor_tree::put( "board.physical.electrical.12v_aux.voltage", m12v_aux_vol ); 
         sensor_tree::put( "board.physical.electrical.12v_aux.current", m12v_aux_curr );
 
+        // U50 has different power rails, so different calculation
+        if (strstr(name(), "_u50_")) {
+            pcidev::get_dev(m_idx)->sysfs_get( "xmc", "xmc_3v3_pex_vol", errmsg, m3v3_pex_vol );
+            pcidev::get_dev(m_idx)->sysfs_get("xmc", "xmc_3v3_pex_curr", errmsg, m3v3_pex_curr);
+            sensor_tree::put( "board.physical.electrical.3v3_pex.voltage", m3v3_pex_vol );
```

Comment:
sensor_tree::put puts the value in xbutil dump. This is duplicating that code. If we refactor this code to have common power calculation, can you please move the other "put"s (12v_pex.vol, 12v_pex.curr, 12v_aux.vol, 12v_aux.curr) to xbutil dump, otherwise xbutil dump and query won't be populated correctly.

Reply:
Moved puts into dump as part of power calc refactor.